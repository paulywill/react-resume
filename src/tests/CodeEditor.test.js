/* eslint-disable */

import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../store';
import CodeEditor from '../components/Tools/CodeEditor';
import AceEditor from 'react-ace';

import { configure, mount, shallow } from 'enzyme';
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });


test('Renders correctly and matches snapshot', () => {
  const tree = renderer
    .create(<Provider store={store}><CodeEditor /></Provider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});


describe('Test updating', () => {
  const domElement = document.getElementById("app");
  const mountOptions = {
    attachTo: domElement
  };

  test("It renders", () => {
    shallow(<Provider store={store}><CodeEditor /></Provider>);
  });
  

  test("Did Update", () => {
    //TO-DO: refactor so there's not so many nested const; reduce testing time
    const startValue = "start value";
    
    const wrapper = mount(<Provider store={store}><CodeEditor/></Provider>);
    let editorValue = wrapper.find(AceEditor).prop('value')
    //wrapper.find(AceEditor).setState({ 'value' : 'bar' });
    //wrapper.update();
    console.log(typeof editorValue); 

    let obj = JSON.parse(editorValue);
    console.log(obj);
    console.log('name: ' +  obj.header.name); 
    obj.header.name = 'Kobe Bryant';
    console.log('name: ' + obj.header.name); 
    editorValue = JSON.stringify(obj);
    console.log(editorValue);
    //wrapper.find(AceEditor).setState({ 'value': editorValue });
   // wrapper.find(AceEditor).setProps('children', <AceEditor 'value': editorValue />)
    wrapper.setProps({ children: 
      <AceEditor
        mode="json"
        theme={'tomorrow_night_bright'}
        name="json-resume-editor"
        enableBasicAutocompletion={false}
        enableLiveAutocompletion={false}
        value={editorValue}
        showLineNumber
        showPrintMargin={false}
        tabSize={3}
        onChange={CodeEditor.onResumeChange}
        editorProps={{
          $blockScrolling: Infinity
        }}
      /> });

    wrapper.update();
    console.log(wrapper.find(AceEditor).prop('value')); 
  });



});
