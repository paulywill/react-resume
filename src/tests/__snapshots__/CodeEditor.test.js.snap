// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly and matches snapshot 1`] = `
<div
  className="json-resume-tool"
  data-testid="CodeEditor"
>
  <div
    className="ui scale down right very wide sidebar"
    style={
      Object {
        "backgroundColor": "#fff",
        "maxHeight": "100vh",
        "maxWidth": "100vw",
        "overflowX": "hidden",
        "width": "100vw",
      }
    }
  >
    <div
      className="ui massive borderless top attached menu"
      style={
        Object {
          "backgroundColor": "#fff",
          "border": "none",
          "maxWidth": "100%",
        }
      }
    >
      <div
        className="left item"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="edit icon"
          onClick={[Function]}
        />
        Code Editor
      </div>
      <div
        className="left item"
        onClick={[Function]}
        style={
          Object {
            "color": "blue",
          }
        }
      >
        waiting for changes
      </div>
      <a
        className="right icon item"
        onClick={[Function]}
        style={
          Object {
            "minWidth": 60,
          }
        }
      >
        <i
          aria-hidden="true"
          className="angle left icon"
          onClick={[Function]}
        />
        
      </a>
    </div>
    <div
      style={
        Object {
          "height": "92%",
          "overflowX": "hidden",
          "paddingBottom": 20,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 10,
        }
      }
    >
      <div
        className="ui blue segment"
        style={
          Object {
            "backgroundColor": "#fff",
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <div
          id="json-resume-editor"
          style={
            Object {
              "height": "500px",
              "width": "500px",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
